[Hub Generated] Review request for Microsoft.ManagedServices to add version preview/2019-04-01-preview

ajay-sainy opened the pull request:
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/app/branch/ajay-sainy/azure-rest-api-specs/dev-managedservices-Microsoft.ManagedServices-2019-04-01-preview...master).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

Commit:
Added new preview version with async apis

Commit:
1. Delete operation (RegistrationDefintion/RegistrationAssignment) on 200 returns empty response. Updated both preview apis.
2. Removed operationstatuses api from specs
3. Updated examples
4. Added 200 response for delete registration assignment

Updated readme
Merge branch 'master' of https://github.com/ajay-sainy/azure-rest-api-specs

Commit:
Updated readme
Merge branch 'master' of https://github.com/ajay-sainy/azure-rest-api-specs

AutorestCI commented:
# Automation for azure-sdk-for-go

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-js

azuresdkci commented:
Can one of the admins verify this patch?

adxsdknet commented:
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6269 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6269/commits/3924ce5368b83a14308e4d58bfab3cc45863e4ae

AutorestCI commented:
# Automation for azure-sdk-for-java

NelsonDaniel reviewed (changes requested):
@ajay-sainy from the log at https://dev.azure.com/azure-sdk/public/_build/results?buildId=30880, it looks like you need to update the readme.go.md to generate under a preview subdirectory

ravbhatnagar commented on specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2019-04-01-preview/managedservices.json:
```diff
@@ -0,0 +1,902 @@
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
```
200 OK response also needs to be modeled. When the long running operation completes, GET on the location header will return 200 OK. the client needs to know what is the final response of this operation. Hence it should be modeled

ajay-sainy replied:
Added 200 also.

ravbhatnagar commented on specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2019-04-01-preview/managedservices.json:
```diff
@@ -0,0 +1,902 @@
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.ManagedServices/operationstatuses/{operationId}": {
```
this does not need to be modeled in swagger. The value comes from the header.

ajay-sainy replied:
Removed

ravbhatnagar reviewed (commented):
This PR is making a few sync operations as async. Thats the only change. A couple of small things to fix. This is good to be approved/merged from ARM side once the couple of comments are addressed.

ajay-sainy committed:
Deleted GetOperationStatuses

Merge branch 'dev-managedservices-Microsoft.ManagedServices-2019-04-01-preview' of https://github.com/ajay-sainy/azure-rest-api-specs into dev-managedservices-Microsoft.ManagedServices-2019-04-01-preview

ajay-sainy committed:
Merge branch 'dev-managedservices-Microsoft.ManagedServices-2019-04-01-preview' of https://github.com/ajay-sainy/azure-rest-api-specs into dev-managedservices-Microsoft.ManagedServices-2019-04-01-preview

ajay-sainy commented:
@ravbhatnagar  I made the changes you suggested. In addition to those changes, there is one more change. As per this change, delete operation for both RegistrationDefinition and RegistrationAssignment now returns empty response with 200. Spec of previous api version (2018-06-01-preview) is also updated.

NelsonDaniel reviewed (changes requested):
@ajay-sainy the file specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2019-04-01-preview/examples/GetOperationStatuses.json is not referenced anywhere; for this reason one of the checks is not passing.

NelsonDaniel commented:
Requesting reassignment because I am OOF.

ajay-sainy committed:
Merge branch 'dev-managedservices-Microsoft.ManagedServices-2019-04-01-preview' of https://github.com/ajay-sainy/azure-rest-api-specs into dev-managedservices-Microsoft.ManagedServices-2019-04-01-preview

sanjaiganesh commented:
>     "version": "2018-06-01-preview",

update version

---
Refers to: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json:5 in 59ef0ec. [](commit_id = 59ef0ecd708afc065ee4b8cc38e65273d5bc89ca, deletion_comment = False)

sanjaiganesh commented on specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2019-04-01-preview/managedservices.json:
```diff
@@ -0,0 +1,855 @@
+  "swagger": "2.0",
+  "info": {
+    "title": "ManagedServicesClient",
+    "version": "2018-06-01-preview",
```
>2018 [](start = 16, length = 4)

update version

ajay-sainy replied:
Done

sanjaiganesh commented on specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2019-04-01-preview/managedservices.json:
```diff
@@ -0,0 +1,855 @@
+      },
+      "description": "Error response."
+    },
+    "OperationStatus": {
```
>OperationStatus [](start = 5, length = 15)

where is this used ?

sanjaiganesh replied:
you missed to remove this after removing operationstatus api

---
In reply to: [289587436](https://github.com/Azure/azure-rest-api-specs/pull/5947#discussion_r289587436) [](ancestors = 289587436)

ajay-sainy replied:
Oh yes, I forgot to remove. Removed now :)

sanjaiganesh commented:
Requested few more minor changes.. and pinging Ajay offline

ajay-sainy committed:
Resolved review comments by Sanjai

1. Updated version
2. Removed Operationstatus

kpajdzik commented on specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2019-04-01-preview/managedservices.json:
```diff
@@ -0,0 +1,845 @@
+            ],
+            "x-ms-enum": {
+              "name": "ProvisioningState",
+              "modelAsString": false
```
Is there any reason why you don't want `modelAsString` set to true? In case you add an entry later, it may be breaking for older generated clients.

ajay-sainy replied:
Updated. Thanks for the suggestion.

kpajdzik commented on specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2019-04-01-preview/managedservices.json:
```diff
@@ -0,0 +1,845 @@
+            ],
+            "x-ms-enum": {
+              "name": "ProvisioningState",
+              "modelAsString": false
```
Same here

ajay-sainy replied:
Updated. Thanks for the suggestion.

kpajdzik commented on specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2019-04-01-preview/managedservices.json:
```diff
@@ -0,0 +1,845 @@
+                  ],
+                  "x-ms-enum": {
+                    "name": "ProvisioningState",
+                    "modelAsString": false
```
and here

ajay-sainy replied:
Updated. Thanks for the suggestion.

kpajdzik reviewed (changes requested):
Take a look at my modelAsString comments. Other than that, looks good :)

ajay-sainy committed:
Changed Modelasstring to true as per the suggestion

kpajdzik commented:
@NelsonDaniel Can you re-review?
